src/main/java/com/mihir/pipposample/Employee/EmployeeDao.java

```diff
@@ -0,0 +1,180 @@
+package com.mihir.pipposample.Employee;
+
+import com.google.inject.Inject;
+import com.mongodb.MongoClient;
+import com.mongodb.MongoClientOptions;
+import com.mongodb.MongoException;
+import com.mongodb.ServerAddress;
+import com.mongodb.client.MongoCollection;
+import com.mongodb.client.MongoDatabase;
+import com.mongodb.event.ServerHeartbeatFailedEvent;
+import com.mongodb.event.ServerHeartbeatStartedEvent;
+import com.mongodb.event.ServerHeartbeatSucceededEvent;
+import com.mongodb.event.ServerMonitorListener;
+import org.bson.Document;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class EmployeeDao implements ServerMonitorListener {
+    private enum HEALTH_STATUS{
+        STARTED, SUCCEEDED, FAILED
+    }
+
+    private HEALTH_STATUS status;
+    private MongoClient mongo;
+    private String mongoHost = "127.0.0.1";
+    private Integer mongoPort = 27017;
+    private String database = "EmployeeDB";
+    private String empCollection = "Employee";
+
+    Object healthCheckLock = new Object();
+
+    private ServerAddress servers;
+
+    @Inject
+    public EmployeeDao(){
+        super();
+
+        servers = new ServerAddress(mongoHost, mongoPort);
+        MongoClientOptions clientOptions = MongoClientOptions.builder()
+                .connectionsPerHost(10)
+                .addServerMonitorListener(this)
+                .build();
+
+        mongo = new MongoClient(servers , clientOptions);
+    }
+
+    public void createEmployee(Document employee) throws Exception{
+        if(status == HEALTH_STATUS.FAILED || mongo == null)
+            throw new Exception("Unable to connect to MongoDao");
+
+        if(!employee.containsKey("email")) throw new Exception("Email is a mandatory field.");
```

Comment:
Never throw generic exception
Create a Dao Exception or a Service Level Exception that extents Exception
check https://github.com/unbxd/banshee/blob/a63ad6a619021c6749c781a965f08f7db5802559/commons/src/main/java/com/unbxd/commons/task/TaskException.java#L15